tofu/geom/_GG02.pyx

```diff
@@ -3330,52 +2965,461 @@ def LOS_calc_signal2(func, double[:,::1] Ds, double[:,::1] us, dL,
         else:
             val = func(np.repeat(Ds,nbrep,axis=1) + k[None,:]*usbis,
                        t=t, **fkwdargs)
-
-        indbis = np.concatenate([0],ind,[k.size])
+        indbis = np.concatenate(([0],ind,[k.size]))
         # Integrate
         if method=='sum':
-            for ii in range(0,ND):
-                sig[:,ii] = np.sum(val[:,indbis[ii]:indbis[ii+1]], axis=-1)*reseff[ii]
+            for ii in range(num_los):
+                sig[:,ii] = np.sum(val[:,indbis[ii]:indbis[ii+1]],
+                                   axis=-1)*reseff[ii]
         elif method=='simps':
-            for ii in range(0,ND):
+            for ii in range(num_los):
```

Comment:
I found that when `range(0,n)` is used instead of `range(n)` cython actually first creates a tuple for the `(0,n)` that is passed later to the range function. Seems like a silly little conversion, so I just avoid indicating the `0`